### ISSUE TITLE
Lot of translation errors after upgrade to Matomo 5.0.1 : “The string ***_*** was not loaded in javascript”

### ISSUE THREAD

**Issue by YorickHRM** [piwik/piwik#21874]
### What happened?

Hi,

I upgraded from Matomo 4.13.3 to Matomo 5.0.1.
There were no errors during file and database update, but now we have a lot of missing french translations on many pages (dashboard, settings, etc).

![capture-matomo](https://github.com/matomo-org/matomo/assets/117072767/d4bfc7f7-1cd7-493a-b278-74955d162818)

In english, everything is fine, all texts are displayed.

This is what I checked :

* File integrity is OK
* All plugins are up to date
* Undisplayed translations are present in the langs/fr.json file of the corresponding plugin

When translations are missing, there is a JS problem on the page.
Here is the error that I got in dev tools :

```
Uncaught SyntaxError: Unexpected token ';' (at index.php?module=UsersManager&action=userSettings&idSite=1&period=day&date=yesterday:67:20)
```

And the code causing the error in index.php :

```JS
<script type="text/javascript">
var translations = ;
if (typeof(piwik_translations) == 'undefined') { var piwik_translations = new Object; }for(var i in translations) { piwik_translations[i] = translations[i];} 
</script>
```
This means $clientSideTranslations is empty in core/Translation/translator.php (line 205).

Do you have any idea what is causing this problem ?

### What should happen?

Texts should all been displayed in french.

### How can this be reproduced?

Upgrade from Matomo 4.13.3 (with default language set to 'fr' in config file), to Matomo 5.0.1.

### Matomo version

5.0.1

### PHP version

7.4.33

### Validations

- [X] Read our [Contributing Guidelines](https://github.com/matomo-org/matomo/blob/5.x-dev/CONTRIBUTING.md).
- [X] Follow our [Security Policy](https://github.com/matomo-org/matomo/blob/5.x-dev/SECURITY.md).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating duplicates.
- [X] The provided steps to reproduce is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the Bug.

**Comment by sgiehl** [comment]
@YorickHRM That's really strange. Looks like for some reason the translations for javascript aren't loaded correctly.
Do you maybe have any custom or third party plugins installed?

**Comment by YorickHRM** [comment]
Hi @sgiehl ,

In total, I have 55 plugins installed and activated.
50 Core, 4 Official and 1 third-party.
All of them are up to date.

I tried to deactivate the non core plugins, and clear the cache, but the problem remains.

I also tried to switch to another language (deutch), and the translations where OK.
The problem is only when I switch to the french language.

The problem still persists after upgrade to 5.0.2.

**Comment by sgiehl** [comment]
Hm. Very strange. Locally that works without problems. Normally the English translations should be used by default and only those available in the chosen language are overwritten. Therefor it's weird, why it does work in one, but not another language.
Does it work correctly in English as well?

**Comment by michalkleiner** [comment]
Might try update to 5.0.2 as well, perhaps it's just some file inconsistency?

**Comment by YorickHRM** [comment]
@sgiehl 
Yes, it works in English. Seems to be only a problem with French language.

@michalkleiner 
I installed 5.0.2 this morning, but the error is still present.
I will try to overwrite all the files with those from the latest bundle (even if System Check says file integrity is OK).

**Comment by sgiehl** [comment]
@YorickHRM If overwriting the files doesn't help as well, please share the list of additional plugins you have installed. Maybe one of them is breaking up something around translations.

**Comment by YorickHRM** [comment]
Problem is still the same after overwriting the files.

Here are the 4 official plugins :

![image](https://github.com/matomo-org/matomo/assets/117072767/a3938e7d-f459-4bce-a048-cfb1dd2a36d6)

And the one from third-party :

![image](https://github.com/matomo-org/matomo/assets/117072767/383daf00-5941-4265-ba03-10d248869b98)

Deactivating them doesn't solve the issue.

**Comment by sgiehl** [comment]
Ok. I had a look at those plugins, but they don't look like they could cause such issues in any way.
Did you already clear the tmp directory of Matomo or tried running `./console cache:clear` ?

**Comment by YorickHRM** [comment]
Yes, I ran `./console cache:clear` almost every time I made changes to the file system just to be sure that no cache file would cause the problem.

**Comment by sgiehl** [comment]
If manually purging the tmp folder of Matomo also doesn't help, I'm out of ideas at the moment

**Comment by YorickHRM** [comment]
No problem, thank you for the help.

I'll try to debug the $clientSideTranslations variable when I have some time.

Maybe something went wrong when upgrading directly from 4.13... 
Should have updated to 4.16 first.

**Comment by michalkleiner** [comment]
How did you update @YorickHRM? Just plain files override/replacement or using the UI?

**Comment by YorickHRM** [comment]
I updated using the UI.

The update button on the dashboard offered to update to **4.16**.
I followed the step to automatically update via UI (as I usualy do when updates are available).

But I didn't notice the small message telling it was going to update to 5.0.1, not 4.16... 🤓 
My bad.

**Comment by michalkleiner** [comment]
Hi @YorickHRM, did you manage to get any more insights into what could be going on? We are out of ideas here and didn't have any other reports of similar behaviour, so it might (unfortunately) be a one off/local issue with your installation that we can't really easily debug remotely.

**Comment by YorickHRM** [comment]
Hi @michalkleiner,

I replaced all the files with the ones of a fresh 5.0.2 installation.
I still have the problem.

Locally, with a clean database, all the french translations are shown.
I couldn't reproduce the problem.

I'm still looking for the cause, but doesn't have much time to spend on Matomo.

**Comment by YorickHRM** [comment]
Found it !

In core/Translation/Translator.php (line 205), the json_encode returned false.
So i looked for the value returned by json_last_error : JSON_ERROR_UTF8.

I looped through the translation array and found the line that couldn't be converted to json : lang/fr.json (line 421)
```
"SaveImageOnYourComputer": "Pour enregistrer l'image sur votre ordinateur, faites un clic droit dessus et sélectionnez «&nbsp;Enregistrer l'image sous…&nbsp;»",
```

I replaced both `&nbsp;` with a simple space, and now the error is gone.

**Comment by michalkleiner** [comment]
Nice find @YorickHRM, thank you for taking the time debugging the issue!
I've updated the translation in Weblate so it will be fixed in Matomo 5.1.0 which is likely a few weeks away. For now I will close the issue as fixed but if you come across again after 5.1.0 is out, please let us know.
